cranelift/filetests/src/function_runner.rs

```diff
@@ -563,6 +623,47 @@ fn make_trampoline(name: UserFuncName, signature: &ir::Signature, isa: &dyn Targ
     func
 }
 
+/// Hostcall invoked directly from a compiled function body to test
+/// exception throws.
+extern "C" fn __cranelift_throw(
+    entry_fp: usize,
+    exit_fp: usize,
+    exit_pc: usize,
+    tag: u32,
+    payload1: usize,
+    payload2: usize,
+) -> ! {
+    let compiled_test_file = unsafe {
+        std::mem::transmute::<*const CompiledTestFile, &CompiledTestFile>(COMPILED_TEST_FILE.get())
+    };
+    let mut unwind_host = cranelift_unwinder::UnwindHost;
+    let module_lookup = |pc| {
+        compiled_test_file
+            .module
+            .as_ref()
+            .unwrap()
+            .lookup_exception_data(pc)
+    };
+    unsafe {
+        match cranelift_unwinder::compute_throw_action(
+            &mut unwind_host,
+            module_lookup,
+            exit_pc,
+            exit_fp,
+            entry_fp,
+            tag,
+        ) {
+            cranelift_unwinder::ThrowAction::Handler { pc, sp, fp } => {
+                cranelift_unwinder::resume_to_exception_handler(pc, sp, fp, payload1, payload2);
+                // Doesn't return.
```

Comment:
It should be ok to skip this comment since the type system basically all guarantees that for us (or perhaps move this beforehand as a lead-in explanation for what the function's doing?)